x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -456,7 +457,7 @@ static RoleDescriptor transformRole(String id, BytesReference sourceBytes, Logge
         try {
             // we pass true as last parameter because we do not want to reject permissions if the field permissions
             // are given in 2.x syntax
-            RoleDescriptor roleDescriptor = RoleDescriptor.parse(name, sourceBytes, true, XContentType.JSON);
+            RoleDescriptor roleDescriptor = RoleDescriptor.parse(name, sourceBytes, true, XContentType.JSON, false);
```

Comment:
We could use an unit test for the `transformRole` method to ensure bytes with `restriction` is rejected.

Reply:
Good suggestion. I'll add a unit test for it.